tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,380 @@
+    port = next(available_ports)
+    random_bytes = data_bytes(10)
+    psk_hash_alg = get_psk_hash_alg_from_cipher(cipher)
+    skip_invalid_psk_tests(provider, psk_hash_alg)
+
+    client_psk_params = setup_s2n_psk_params(psk_identity, psk_secret, psk_hash_alg)
+    client_options = setup_provider_options(S2N.ClientMode, port, cipher, curve, None, random_bytes, client_psk_params)
+
+    if provider == S2N:
+        server_psk_params = setup_s2n_psk_params(psk_identity, psk_secret, psk_hash_alg)
+    else:
+        server_psk_params = setup_openssl_psk_params(psk_identity, psk_secret)
+        server_psk_params += [ '-nocert' ]
+    server_options = setup_provider_options(provider.ServerMode, port, cipher, curve, None, None, server_psk_params)
+
+    server = managed_process(provider, server_options, timeout=5, close_marker=str(random_bytes))
+    client = managed_process(S2N, client_options, timeout=5)
+
+    for results in client.get_results():
+        results.assert_success()
+        validate_negotiated_psk_s2n(Outcome.psk_connection, psk_identity, results)
+
+    for results in server.get_results():
+        results.assert_success()
+        if provider == S2N:
+            validate_negotiated_psk_s2n(Outcome.psk_connection, psk_identity, results)
+        else:
+            validate_negotiated_psk_openssl(Outcome.psk_connection, results)
+        assert random_bytes in results.stdout
+
+
+"""
+Tests S2N client's behavior with multiple PSKs and no fallback option. 
+
+Note that OpenSSL does not support multiple PSKs. 
+"""
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", PSK_PROVIDERS, ids=get_parameter_name)
+@pytest.mark.parametrize("psk_identity", PSK_IDENTITY_LIST, ids=get_parameter_name)
+@pytest.mark.parametrize("psk_secret", PSK_SECRET_LIST, ids=get_parameter_name)
+def test_s2n_client_multiple_psks(managed_process, cipher, curve, protocol, provider, psk_identity, psk_secret):
+    port = next(available_ports)
+    random_bytes = data_bytes(10)
+    psk_hash_alg = get_psk_hash_alg_from_cipher(cipher)
+    skip_invalid_psk_tests(provider, psk_hash_alg)
+
+    client_psk_params = setup_s2n_psk_params(psk_identity, psk_secret, psk_hash_alg)
+    client_psk_params.extend(setup_s2n_psk_params(PSK_IDENTITY_NO_MATCH, PSK_SECRET_NO_MATCH, psk_hash_alg))
+    client_options = setup_provider_options(S2N.ClientMode, port, cipher, curve, None, random_bytes, client_psk_params)
+
+    server_psk_params = []
+    if provider == OpenSSL:
+        """
+        OpenSSL Provider does not support multiple PSKs in the same connection, 
+        the last psk params is the final psk used in the connection. 
+        """
```

Comment:
If OpenSSL doesn't actually support this use case, should it be a provider for this test?

Reply:
Yes - as we test the behavior when multiple PSKs are sent from s2n to OpenSSL.